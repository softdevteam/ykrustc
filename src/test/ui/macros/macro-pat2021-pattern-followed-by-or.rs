#![feature(edition_macro_pats)]
#![allow(unused_macros)]
macro_rules! foo { ($x:pat2021 | $y:pat2021) => {} } //~ ERROR `$x:pat2021` is followed by `|`, which is not allowed for `pat2021` fragments
macro_rules! baz { ($x:pat2015 | $y:pat2015) => {} } // should be ok
macro_rules! qux { ($x:pat2015 | $y:pat2021) => {} } // should be ok
macro_rules! ogg { ($x:pat2021 | $y:pat2015) => {} } //~ ERROR `$x:pat2021` is followed by `|`, which is not allowed for `pat2021` fragments
macro_rules! match_any {
    ( $expr:expr , $( $( $pat:pat2021 )|+ => $expr_arm:pat2021 ),+ ) => { //~ ERROR  `$pat:pat2021` may be followed by `|`, which is not allowed for `pat2021` fragments
        match $expr {
            $(
                $( $pat => $expr_arm, )+
            )+
        }
    };
}

fn main() {
    let result: Result<i64, i32> = Err(42);
    let int: i64 = match_any!(result, Ok(i) | Err(i) => i.into());
    assert_eq!(int, 42);
}
