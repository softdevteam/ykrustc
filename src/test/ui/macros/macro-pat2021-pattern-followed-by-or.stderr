error: `$x:pat2021` is followed by `|`, which is not allowed for `pat2021` fragments
  --> $DIR/macro-pat2021-pattern-followed-by-or.rs:3:32
   |
LL | macro_rules! foo { ($x:pat2021 | $y:pat2021) => {} }
   |                                ^ not allowed after `pat2021` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `if` or `in`

error: `$x:pat2021` is followed by `|`, which is not allowed for `pat2021` fragments
  --> $DIR/macro-pat2021-pattern-followed-by-or.rs:6:32
   |
LL | macro_rules! ogg { ($x:pat2021 | $y:pat2015) => {} }
   |                                ^ not allowed after `pat2021` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `if` or `in`

error: `$pat:pat2021` may be followed by `|`, which is not allowed for `pat2021` fragments
  --> $DIR/macro-pat2021-pattern-followed-by-or.rs:8:40
   |
LL |     ( $expr:expr , $( $( $pat:pat2021 )|+ => $expr_arm:pat2021 ),+ ) => {
   |                                        ^ not allowed after `pat2021` fragments
   |
   = note: allowed there are: `=>`, `,`, `=`, `if` or `in`

error: aborting due to 3 previous errors

